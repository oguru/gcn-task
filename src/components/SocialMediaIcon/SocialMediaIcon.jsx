import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFacebookF } from '@fortawesome/free-brands-svg-icons'
import { faInstagram } from '@fortawesome/free-brands-svg-icons';
import { faTwitter } from '@fortawesome/free-brands-svg-icons';
import React from "react";
import styles from "./SocialMediaIcon.module.scss";


const SocialMediaIcon = (props) => {
  const {isPrimary, icon, link} = props;

  const checkIcon = () => {
    switch (icon) {
      case "twitter":
        return faTwitter;
      case "facebook":
        return faFacebookF;
      default:
        return faInstagram;
    }
  }

  const mediaIconStyle = isPrimary ? 
    `${styles.mediaIconCont} ${styles.navIcon}` 
    : 
    `${styles.mediaIconCont} ${styles.footerIcon}`;

  return (
    <>
      <a 
        href={link} 
        className={mediaIconStyle} 
        target="blank"
      >
        <FontAwesomeIcon 
          className={styles.mediaIcon} 
          icon={checkIcon()}
        />
      </a>
    </>
  );
};

export default SocialMediaIcon;
