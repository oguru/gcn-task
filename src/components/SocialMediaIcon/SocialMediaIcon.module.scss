@import "../../variables.scss";

.mediaIconCont {
    align-items: center;
    border-radius: $base-spacing/2;
    color: white;
    display: flex;
    height: 30px;
    justify-content: center;
    margin: 0 4px;
    transition: 200ms;
    width: 30px;

    &:hover {
        background-color: $primary-color;
    }

    .mediaIcon {
        font-size: 20px;
    }
}

.footerIcon {
    background-color: white;
    color: black;
}

.navIcon {
    height: 25px;
    width: 25px;

    .mediaIcon {
        font-size: 15px;
    }
}