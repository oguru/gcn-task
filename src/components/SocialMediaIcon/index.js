import SocialMediaIcon from "./SocialMediaIcon";

export default SocialMediaIcon;
