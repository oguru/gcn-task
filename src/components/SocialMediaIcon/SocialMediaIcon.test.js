import { cleanup, render } from "@testing-library/react";
import { mount } from "enzyme";
import React from "react";

import SocialMediaIcon from "./SocialMediaIcon";

describe("SocialMediaIcon tests", () => {
  let primaryFbComponent;
  let secondaryIgComponent;
  const link = "testlink";
  
    beforeEach(() => {
      primaryFbComponent = mount(
        <SocialMediaIcon 
          isPrimary={true} 
          icon={"facebook"}
          key="facebookn" 
          link={link}
        />
      );
  
        secondaryIgComponent = mount(
          <SocialMediaIcon 
          isPrimary={false} 
          icon={"instagram"} 
          key="instagramf" 
          link={link}
        />        
      );
    });

  afterEach(cleanup);

  it("should render", () => {
    expect(render(primaryFbComponent)).toBeTruthy();
    expect(render(secondaryIgComponent)).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(primaryFbComponent).toMatchSnapshot();
    expect(secondaryIgComponent).toMatchSnapshot();
  });

  it("should contain an svg", () => {
    expect(primaryFbComponent.find("svg").length).toEqual(1);
    expect(secondaryIgComponent.find("svg").length).toEqual(1);
  });

  it("should get the correct css styles based on isPrimary prop", () => {
    expect(primaryFbComponent.find(".navIcon").length).toEqual(1);
    expect(primaryFbComponent.find(".footerIcon").length).toEqual(0);
    expect(secondaryIgComponent.find(".footerIcon").length).toEqual(1);
    expect(secondaryIgComponent.find(".navIcon").length).toEqual(0);
  });

  it("should return the correct icon based on the icon prop", () => {
    expect(primaryFbComponent.find(".fa-facebook-f").length).toEqual(1);
    expect(primaryFbComponent.find(".fa-twitter").length).toEqual(0);
    expect(secondaryIgComponent.find(".fa-instagram").length).toEqual(1);
    expect(secondaryIgComponent.find(".fa-facebook-f").length).toEqual(0);
  });
  
});
