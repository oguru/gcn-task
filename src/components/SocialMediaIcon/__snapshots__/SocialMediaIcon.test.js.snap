// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaIcon tests should match the snapshot 1`] = `ReactWrapper {}`;

exports[`SocialMediaIcon tests should match the snapshot 2`] = `ReactWrapper {}`;
